MainRepository/Program.cs

```diff
@@ -1,7 +1,56 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿var pq = new PriorityQueue<int> ();
+pq.EnQueue (25);
+pq.EnQueue (8);
+Console.WriteLine (pq.DeQueue ());
+pq.EnQueue (27);
+pq.EnQueue (3);
+Console.WriteLine (pq.DeQueue ());
+pq.EnQueue (1);
+pq.EnQueue (2);
+pq.EnQueue (7);
+foreach (var item in pq.mData)
+   Console.Write (item + " ");
+Console.WriteLine ();
+
+/// <summary>Implementation of PriroityQueue using Binary Min Heap</summary>
+public class PriorityQueue<T> where T : IComparable<T> {
+
+   public List<T> mData = new();
+
+   /// <summary>Adds element in the priority queue</summary>
+   public void EnQueue (T item) {
+      mData.Add (item);
+      int ci = mData.Count - 1;
+      while (ci > 0) {
+         int pi = (ci - 1) / 2;
+         if (mData[ci].CompareTo (mData[pi]) >= 0) break;
+         (mData[pi], mData[ci]) = (mData[ci], mData[pi]);
+         ci = pi;
+      }
+   }
+
+   /// <summary>Removes element in the priority queue</summary>
+   public T DeQueue () {
+      int li = mData.Count - 1;
+      T top = mData[0];

```

Comment:
You have used zeroth index element, but in the video it is suggested to skip zeroth index and start from 1 so that the 2*n and 2*n+1 works properly. Please watch the video again.